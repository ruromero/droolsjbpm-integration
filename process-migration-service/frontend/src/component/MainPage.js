import React, { Component } from "react";
import { MenuItem, Icon } from "patternfly-react";

import MigrationPlans from "./tabMigrationPlan/MigrationPlans";
import MigrationDefinitions from "./tabMigration/MigrationDefinitions";
import KieServerClient from "../clients/kieServerClient";
import { ALERT_TYPE_ERROR } from "patternfly-react/dist/js/components/Alert/AlertConstants";
import Notification from "./Notification";
import {
  Masthead,
  MastheadDropdown
} from "patternfly-react/dist/js/components/Masthead";
import MastheadCollapse from "patternfly-react/dist/js/components/Masthead/MastheadCollapse";
import {
  HorizontalNavMenu,
  HorizontalNav,
  HorizontalNavMenuItem
} from "patternfly-react/dist/js/components/HorizontalNav";

export default class MainPage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      kieServerId: "",
      title: "KIE Server Name",
      menuItems: [],
      errorMsg: "",
      showMigrations: false,
      showPlans: true
    };
  }

  componentDidMount() {
    this.loadKieServers();
  }

  showMigrations = () => {
    this.setState({
      showMigrations: true,
      showPlans: false
    });
  };

  showPlans = () => {
    this.setState({
      showMigrations: false,
      showPlans: true
    });
  };

  setErrorMsg = errorMsg => this.setState({ errorMsg });

  resetErrorMsg = () => this.setErrorMsg("");

  loadKieServers = () => {
    KieServerClient.getKieServers()
      .then(res => this.populateKieServers(res))
      .catch(error => {
        this.populateKieServers([]);
        this.setErrorMsg(error);
      });
  };

  populateKieServers = kieServers => {
    let menuItems = [];
    if (kieServers.size === 0) {
      menuItems.push(
        <MenuItem key={0} eventKey="no-kieservers" disabled>
          No configured KieServers
        </MenuItem>
      );
    } else {
      kieServers.map((kieServer, i) => {
        const kieServerId =
          kieServer.id === null ? "unknown-" + i : kieServer.id;
        if (this.state.kieServerId === "") {
          this.setState({ kieServerId });
        }
        menuItems.push(
          <MenuItem
            key={i}
            eventKey={kieServerId}
            onSelect={this.handleChange}
            disabled={kieServer.status !== "SUCCESS"}
          >
            {kieServerId}
          </MenuItem>
        );
      });
    }
    menuItems.push(<MenuItem key="-" divider />);
    menuItems.push(
      <MenuItem
        key="refresh"
        title="Refresh servers list"
        onSelect={this.loadKieServers}
      >
        <Icon type="fa" name="refresh" />
        &nbsp; Refresh
      </MenuItem>
    );
    this.setState({ menuItems });
  };

  handleChange = option => {
    this.setState({ kieServerId: option });
  };

  render() {
    const notification = (
      <Notification
        type={ALERT_TYPE_ERROR}
        message={this.state.errorMsg}
        onDismiss={() => this.setState({ errorMsg: "" })}
      />
    );
    return (
      <React.Fragment>
        <HorizontalNav>
          <Masthead title="Process instance migration" thin navToggle>
            <MastheadCollapse>
              <MastheadDropdown
                id="kieservers-dropdown"
                title={
                  <span>
                    <Icon type="fa" name="server" />
                    &nbsp;
                    {this.state.kieServerId}
                  </span>
                }
              >
                {this.state.menuItems}
              </MastheadDropdown>
            </MastheadCollapse>
          </Masthead>
          {this.state.errorMsg && notification}
          <HorizontalNavMenu>
            <HorizontalNavMenuItem
              id="plans"
              title="Migration plans"
              onItemClick={this.showPlans}
            />
            <HorizontalNavMenuItem
              id="migrations"
              title="Migrations"
              onItemClick={this.showMigrations}
            />
          </HorizontalNavMenu>
        </HorizontalNav>
        <div id="pim-content">
          {this.state.showPlans && (
            <MigrationPlans
              kieServerId={this.state.kieServerId}
              onError={this.setErrorMsg}
            />
          )}
          {this.state.showMigrations && (
            <MigrationDefinitions onError={this.setErrorMsg} />
          )}
        </div>
      </React.Fragment>
    );
  }
}
