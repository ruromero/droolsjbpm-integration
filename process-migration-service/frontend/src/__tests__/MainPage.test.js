import React from "react";
import renderer from "react-test-renderer";

import MainPage from "../component/MainPage";

describe("Main Page", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(<MainPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
