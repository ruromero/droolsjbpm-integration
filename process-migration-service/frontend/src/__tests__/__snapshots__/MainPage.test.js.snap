// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page renders without crashing 1`] = `
Array [
  <nav
    className="navbar navbar-default navbar-pf"
  >
    <nav
      className="navbar navbar-default navbar-pf"
    >
      <div
        className="navbar-header"
      >
        <button
          className="navbar-toggle"
          onClick={[Function]}
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <span
            aria-hidden="true"
            className="icon-bar"
          />
          <span
            aria-hidden="true"
            className="icon-bar"
          />
          <span
            aria-hidden="true"
            className="icon-bar"
          />
        </button>
        <a
          className="navbar-brand"
          href="#"
          onClick={[Function]}
          role="button"
        >
          
          
          Process instance migration
        </a>
      </div>
      <nav
        className="collapse navbar-collapse"
      >
        <ul
          className="nav navbar-nav navbar-right navbar-iconic navbar-utility"
        >
          <li
            className="dropdown"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="nav-item-iconic dropdown-toggle btn btn-link"
              disabled={false}
              id="kieservers-dropdown"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              type="button"
            >
              <span>
                <span
                  aria-hidden={true}
                  className="fa fa-server"
                />
                 
                
              </span>
               
              <span
                className="caret"
              />
            </button>
            <ul
              aria-labelledby="kieservers-dropdown"
              className="dropdown-menu"
              role="menu"
            />
          </li>
        </ul>
      </nav>
    </nav>
    
    <div
      className="collapse navbar-collapse navbar-collapse-1"
    >
      <ul
        className="nav navbar-nav navbar-primary"
      >
        <li
          className=""
          id="plans"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            Migration plans
          </a>
        </li>
        <li
          className=""
          id="migrations"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            Migrations
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <div
    id="pim-content"
  >
    
    <br />
    <div
      className="row"
    >
      <div
        className="col-xs-9"
      >
        <div>
          <label
            htmlFor="filter"
          >
            Filter by Source Container ID: 
          </label>
           
          <input
            id="inputFilter"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="col-xs-3"
      >
        <div
          className="pull-right"
        >
          <button
            className="btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Import
          </button>
           
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Add Plan
          </button>
        </div>
      </div>
    </div>
    <br />
    <table
      className="table table-striped table-hover"
    >
      <thead>
        <tr>
          <th
            className=""
          >
            ID
          </th>
          <th
            className=""
          >
            Plan Name
          </th>
          <th
            className=""
          >
            Description
          </th>
          <th
            className=""
          >
            Source container
          </th>
          <th
            className=""
          >
            Source process
          </th>
          <th
            className=""
          >
            Target container
          </th>
          <th
            className=""
          >
            Target process
          </th>
          <th
            aria-label="Actions"
            className=""
            colSpan={4}
            rowSpan={1}
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;
